samcli/commands/local/invoke/cli.py

```diff
@@ -198,5 +198,5 @@ def _get_event(event_file_name):
 
     # click.open_file knows to open stdin when filename is '-'. This is safer than manually opening streams, and
     # accidentally closing a standard stream
-    with click.open_file(event_file_name, "r") as fp:
+    with click.open_file(event_file_name, "r", encoding="utf-8") as fp:
```

Comment:
This out of scope for this PR, but do you have any idea on how we can detect this in Unit test and not integration test?

Reply:
Running the UTs on Windows is one way (but not necessarily sustainable). Another way is probably forcing the default encoding to be `cp1252` during UTs, so that it simulates the Windows case.

Reply:
Setting `LC_ALL=C PYTHONUTF8=0` seems to be one way to force a single-byte default encoding.